crowbar_framework/app/models/service_object.rb

```diff
@@ -1190,6 +1193,15 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["deployment"][@bc_name]["elements"]["to_remove"] ||= {}
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] ||= []
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] << elem_remove
```

Comment:
`to_remove` isn't supposed to reference the whole databag, just a small part of it.  Why would that matter?  Whether you write `to_remove` or `databag["deployment"][@bc_name]["elements"]["to_remove"]`, shouldn't you get the same object every time?